internal/client/api/backend.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package api
+
+import (
+	"sync"
+
+	"github.com/ChainSafe/gossamer/internal/primitives/blockchain"
+	"github.com/ChainSafe/gossamer/internal/primitives/core/offchain"
+	"github.com/ChainSafe/gossamer/internal/primitives/runtime"
+	statemachine "github.com/ChainSafe/gossamer/internal/primitives/state-machine"
+	"github.com/ChainSafe/gossamer/internal/primitives/storage"
+)
+
+// State of a new block.
+type NewBlockState uint
```

Comment:
Why not to use the smaller uint? for example uint8?

```suggestion
type NewBlockState uint8
```

Reply:
Are we planning to change it or are you going to keep uint? 🤔

Reply:
I changed it. https://github.com/ChainSafe/gossamer/blob/83edb1e7bc65b5c1d569be8e9d5acaea891a3531/internal/client/api/backend.go#L17

Reply:
Oh sorry! It wasn't updated on my side! lgtm